android-build-tools/src/apk.rs

```diff
@@ -131,6 +133,47 @@ impl<'a> UnalignedApk<'a> {
         Ok(())
     }
 
+    pub fn add_lib_recursively(
```

Comment:
Rust supports multiple impl blocks for struct so it can be stay decomposed (as readelf.rs) to keep code clear.

Reply:
well I don't think that works accross files but I might be wrong. if the reader has to look for the method because it's somewhere else, I'm not sure that's an improvement

Reply:
It works and I use it to decompose structs with huge number of functions.